src/pdc-api.ts

```diff
@@ -13,28 +13,38 @@ const throwNotOk = (res: Response): Response => {
   return res;
 };
 
-const logError = (error: unknown, path: string) => {
-  logger.error({ error }, `Error fetching ${path}`);
+const logError = (error: unknown, path: string, params: URLSearchParams) => {
+  logger.error({ error, params }, `Error fetching ${path}`);
 };
 
 // Custom React hook to make authenticated requests to the configured API
-const usePdcApi = <T>(path: string): T | null => {
+const usePdcApi = <T>(
+  path: string,
+  params: URLSearchParams = new URLSearchParams(),
+): T | null => {
   const { fetch } = useOidcFetch();
   const [response, setResponse] = useState(null);
 
   useEffect(() => {
-    fetch(new URL(path, API_URL))
+    const url = new URL(path, API_URL);
+    url.search = params.toString();
+    fetch(url)
       .then(throwNotOk)
       .then((res) => res.json())
       .then(setResponse)
-      .catch((e) => logError(e, path));
+      .catch((e) => logError(e, path, params));
 
     /* eslint-disable-next-line react-hooks/exhaustive-deps --
+     *
      * fetch should not be a dependency, because although it or its internal
      * state may change from render to render, such changes are not relevant:
      * a change in the way we make a request should not trigger an API request
+     *
+     * params is a dependency, but as an object - and often a newly-created
+     * object - depending on it directly causes spurious renders; instead, use
+     * its string value, which should be stable
```

Comment:
Wow, good eye. Not sure I'd have realized this is what was going on!

Reply:
I noticed it when my network tab starting going nuts! See also [Removing unnecessary object dependencies ](https://react.dev/reference/react/useEffect#removing-unnecessary-object-dependencies).